lib/Sema/CSApply.cpp

```diff
@@ -7855,7 +7855,8 @@ bool ConstraintSystem::applySolutionFix(
       (!resolved->getPath().empty() &&
        fix.first.getKind() != FixKind::ExplicitlyEscaping &&
        fix.first.getKind() != FixKind::ExplicitlyEscapingToAny &&
-       fix.first.getKind() != FixKind::AddConformance))
+       fix.first.getKind() != FixKind::AddConformance &&
+       fix.first.getKind() != FixKind::TreatRValueAsLValue))
```

Comment:
It looks like this is on the threshold where a 'switch' statement starts to make sense

Reply:
Right, I think I can remove this logic altogether...